### PR TITLE
deployment: consolidate environments and module config

### PR DESCRIPTION
**Pull request by matz-e**
Moves the part of the module configuration that is specific to the
environment into the environment definition:

* simplifies our setup
* now there is one global module config (blacklisting everything)
* whitelists are in the same file as specs to build

Otherwise, there are very few changes:
* adjustment of the ruby script that does the user-facing whitelist
* `compilers.yaml` with RH compilers is in `~/.spack/linux` to exclude those from the regular compilers

### CODE REVIEW

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

Another purge of the installation directory

**Comment by matz-e** [comment]
Retest this please

Another purge of the installation directory

**Comment by matz-e** [comment]
@pramodk would be nice to get this in… would simplify refactoring quite a bit!

File: deploy/configs/compilers.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      cc: /usr/bin/gcc
+      cxx: /usr/bin/g++
+      f77: /usr/bin/gfortran
+      fc: /usr/bin/gfortran
+    flags: {}
+    operating_system: rhel7
+    target: x86_64
+    environment: {}
+    extra_rpaths: []
+    modules: []
```
Comment (pramodk) [inline deploy/configs/compilers.yaml:27]: Is this new one / needed ? (spack compiler find should generate this, right?)

**Review by pramodk — APPROVED** [review]
Nothing specific that I can notice as problematic, LGTM!

File: deploy/configs/compilers.yaml (hunk 1 above)
Comment (matz-e) [reply deploy/configs/compilers.yaml:27]: Yes… but I also wanted to have this as a central configuration to build on.

**Pull request merged**